src/System.Private.CoreLib/shared/System/Threading/ThreadPool.cs

```diff
@@ -865,8 +966,9 @@ private static void WaitOrTimerCallback_Context(object state, bool timedOut)
         }
 
         // call back helper
-        internal static void PerformWaitOrTimerCallback(_ThreadPoolWaitOrTimerCallback helper, bool timedOut)
+        internal static void PerformWaitOrTimerCallback(object state, bool timedOut)
         {
+            _ThreadPoolWaitOrTimerCallback helper = (_ThreadPoolWaitOrTimerCallback)state;
```

Comment:
The CoreRT version without cast looked better.

Reply:
It did... but I don't know how to convince the CoreCLR binder to properly bind to it from the unmanaged code.

Reply:
You would need to define a signature for it in https://github.com/dotnet/coreclr/blob/master/src/vm/metasig.h . It should be something like `DEFINE_METASIG_T(SM(_ThreadPoolWaitOrTimerCallback_Bool_RetVoid, C(TPWAITORTIMER_HELPER) F, v))` and then replace `SM_Obj_Bool_RetVoid` here with `SM__ThreadPoolWaitOrTimerCallback_Bool_RetVoid` to refer to it.

Reply:
Thanks! I will give it a try.